### PR TITLE
Update Docker_Publish action to use official images

### PR DESCRIPTION
**Pull request by bathina2**

### CODE REVIEW

File: .github/workflows/docker-publish.yml [hunk 1]
```diff
@@ -9,9 +9,12 @@ on:
     # Publish `v1.2.3` tags as releases.
     tags:
       - v*
+  pull_request:
+    branches: [ master ]
```
Comment (julio-lopez) [inline .github/workflows/docker-publish.yml:13]: Do you want to create a new image on each PR? or is this for testing this PR?

File: .github/workflows/docker-publish.yml [hunk 2]
```diff
@@ -23,30 +26,28 @@ jobs:
...
+      - name: Extract Docker metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      # Build and push Docker image with Buildx (don't push on PR)
+      # https://github.com/docker/build-push-action
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
```
Comment (julio-lopez) [inline .github/workflows/docker-publish.yml:48]: nit: There may be a "versioned" version of this action. This is fine though.

File: .github/workflows/docker-publish.yml (hunk 1 above)
Comment (bathina2) [reply .github/workflows/docker-publish.yml:13]: @julio-lopez It won't actually push the image see line 51 ` push: ${{ github.event_name != 'pull_request' }}`

Comment (bathina2) [reply .github/workflows/docker-publish.yml:13]: But it will try to build the image and validate the Dockerfile

**Pull request merged**